public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -276,28 +276,32 @@ export const Explorer = ({
       queryToRun,
     }: any = historyFromRedirection.location.state;
     batch(() => {
-      dispatch(
-        updateSearchMetaData({
-          tabId,
-          data: {
-            datasources: [
-              {
-                label: datasourceName,
-                type: datasourceType,
-                value: datasourceName,
-                name: datasourceName,
-              },
-            ],
-          },
-        })
-      );
-      dispatch(
-        changeQuery({
-          tabId,
-          query: { [RAW_QUERY]: queryToRun },
-        })
-      );
-      setTempQuery(queryToRun);
+      if (datasourceName && datasourceType) {
+        dispatch(
```

Comment:
Another question out of curiosity: what's the benefit of `dispatch` over `useEffect`? I'm not sure what'd motivate me to use one over the other, I haven't looked deeply into redux.

Reply:
the extent of my knowledge is that dispatch would be used for updating redux state, i'm not sure about if useEffect could be used for the same thing